sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtilsQuarkusTest.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.stargate.sgv2.docsapi.service.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.stargate.bridge.grpc.Values;
+import io.stargate.sgv2.docsapi.DocsApiTestSchemaProvider;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import io.stargate.sgv2.docsapi.service.common.model.RowWrapper;
+import io.stargate.sgv2.docsapi.testprofiles.MaxDepth4;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.inject.Inject;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@QuarkusTest
+@TestProfile(MaxDepth4.class)
+class DocsApiUtilsQuarkusTest {
```

Comment:
do you think it's not possible to pull the other DocsApi test here?

Reply:
Yea MockitoExtension won t work with QuarkusTest. Opened the ticket there, said this is not supported and nor will be due to the classloading they do for tests.

Reply:
Ok, that shouldn't be a big problem in practice, once you start using QuarkusTest you have the whole context and you're less likely to mock things.
And in any case it's just the extension that doesn't work, mocks can still be created explicitly with `Mockito.mock`, or MockitoAnnotations.openMocks` if we want to keep the annotations.

Reply:
you have the `@InjectMock` support directly from Quarkus, you can mock any bean.. So the problem would only be in auto-mock-init non-bean classes or captors..